From 1cf42471407537998dafaadb5aaa2f70afea7a76 Mon Sep 17 00:00:00 2001
From: Dinesh Kumar Mirche <dmirche@habana.ai>
Date: Tue, 10 Jun 2025 14:47:07 +0300
Subject: [PATCH] [SW-227362] g2: sec: make nic txb registers privilege

This patch configures following registers privilege from
non privilege
H6_PROJECT ➔ NIC0_TXB ➔ RL_PORT_EN
H6_PROJECT ➔ NIC0_TXB ➔ RL_PORT_SAT
H6_PROJECT ➔ NIC0_TXB ➔ RL_PORT_RST
H6_PROJECT ➔ NIC0_TXB ➔ RL_PORT_TIMEOUT

This is required to use port RL to constrain SerDes IO BW
to meet US Government restriction

Change-Id: I83049342c966f5382f2399342852011b919a6f43
Signed-off-by: Dinesh Kumar Mirche <dmirche@habana.ai>
(cherry picked from commit 585938da3a9612ded8707bde739a8747eaf7c6dc)
---

diff --git a/drivers/misc/hl_security/gaudi2/gaudi2_protbits_configs.h b/drivers/misc/hl_security/gaudi2/gaudi2_protbits_configs.h
index 2c4164c..2ded869 100644
--- a/drivers/misc/hl_security/gaudi2/gaudi2_protbits_configs.h
+++ b/drivers/misc/hl_security/gaudi2/gaudi2_protbits_configs.h
@@ -462,7 +462,7 @@
 		0x100001ff, 0x60000 }
 
 #define GAUDI2_PROTBITS_DATA112 { \
-		 0xff00000f, 0xff10ffff, 0x80ff, 0x60000 }
+		 0xff00000f, 0x10ffff, 0x8000, 0x60000 }
 
 #define GAUDI2_PROTBITS_DATA113 { \
 		 0xffffff, 0xffff0000, 0x3fd50000, 0xffffff, 0xffff0000, 0x3fd50000, \
